camel-chimera-graph/src/main/java/com/cefriel/operations/GraphSparql.java

```diff
@@ -0,0 +1,93 @@
+        EndpointParams(GraphBean operationConfig) {
+            this(operationConfig.getQuery(),
+                    operationConfig.getChimeraResource(),
+                    operationConfig.getDumpFormat());
+        }
+        EndpointParams {
+            Objects.requireNonNull(outputFormat, "format for sparql query result must be specified using dumpFormat endpoint option");
+
+            if (!validOutputFormats.contains(outputFormat))
+                throw new IllegalArgumentException("unsupported format " + outputFormat + ", must be one of: " + validOutputFormats);
+
+            if (literalQuery == null&& resourceQuery == null)
+                throw new NullPointerException("both sparql queries query cannot be null");
+        }
+    }
+    private record OperationParams (Exchange exchange, RDFGraph graph, String query, String outputFormat) {
+        OperationParams(Exchange exchange, EndpointParams endpointParams) throws Exception {
+            this(exchange,
+                    exchange.getMessage().getBody(RDFGraph.class),
+                    Utils.resolveQuery(endpointParams.literalQuery, endpointParams.resourceQuery, exchange),
+                    endpointParams.outputFormat);
+        }
+
+        OperationParams {
+            Objects.requireNonNull(exchange, "Incoming exchange cannot be null");
+            Objects.requireNonNull(graph, "RDFGraph cannot be null");
+
+            if (query.isBlank())
+                throw new IllegalArgumentException("SPARQL query cannot be blank");
+        }
+    }
+    public static void graphSparql(Exchange exchange, GraphBean operationConfig) throws Exception {
+        EndpointParams endpointParams = new EndpointParams(operationConfig);
+        OperationParams operationParams = new OperationParams(exchange, endpointParams);
+        graphSparql(exchange, operationParams.graph, operationParams.query, operationParams.outputFormat);
+    }
+    private static void graphSparql (Exchange exchange, RDFGraph graph, String query, String outputFormat) {
+        Repository repo = graph.getRepository();
+        RepositoryConnection connection = repo.getConnection();
+        TupleQuery tupleQuery = connection.prepareTupleQuery(query);
+        LOG.info("Executing sparql query...");
+
+        if (outputFormat.equals("memory")) {
+            try (TupleQueryResult result = tupleQuery.evaluate()) {
+                // this is needed because TupleQueryResult is lazy and keeps a connection open to the repository.
+                // we want to get the results and then not care about the repository.
+                List<BindingSet> resultList = QueryResults.asList(result);
+                exchange.getMessage().setBody(resultList, List.class);
+            }
+        }
+        else {
+            StringWriter stringWriter = new StringWriter();
+            switch (outputFormat) {
+                case "json" -> tupleQuery.evaluate(new SPARQLResultsJSONWriter(stringWriter));
```

Comment:
As it is currently implemented the outputFormat is a non-optional parameter, meaning that it MUST be specified and cannot be null, which is the same as being not specified. If it is specified then it MUST be on of the supported formats. I do not know what a sensible value for a default would be here, it is highly dependent on the application. Perhaps a good idea would be to document somewhere that the 'memory' format is possibly the best one if further (Java) operations are needed on the results while the others are more 'final' formats.

Reply:
My suggestion would be to make the parameter optional and use what is now `memory` as the default value. I would expect to be able to get "something" without specifying the parameter, also I do not like the fact that you should specify "memory" as a format... In this way, `outputFormat` can also be changed as `serializationFormat` or something similar, in the sense that it MUST be added if you expect to get a serialisation within the body instead of the in-memory object